docs/map.rst

```diff
@@ -0,0 +1,602 @@
+Maps
+====
+
+.. highlight:: haskell
+
+Maps (sometimes referred to as dictionaries in other languages) allow you to
+store associations between *unique keys* and *values*. There are three
+implementations provided by the ``containers`` package:
+:haddock:`containers/Data.Map.Strict`, :haddock:`Data.Map.Lazy`, and
+:haddock:`Data.IntMap`. You almost never want the lazy version so use
+``Data.Map.Strict``, or if your keys are ``Int`` use ``Data.IntMap``; all of
+these implementations are *immutable*.
+
+::
+
+    data Map k v = ...
+
+    data IntMap v = ...
+
+.. IMPORTANT::
+   ``Map`` relies on the key type ``k`` having instances of the ``Eq`` and
+   ``Ord`` typeclass for its internal representation. These are already defined
+   for builtin types, and if you are using your own data type you can use the
+   `deriving
+   <https://en.wikibooks.org/wiki/Haskell/Classes_and_types#Deriving>`_
+   mechanism.
+
+
+Short Example
+-------------
+
+The following GHCi session shows some of the basic map functionality::
+
+    import qualified Data.Map.Strict as Map
+
+    let nums = Map.fromList [(1,"one"), (2,"two"), (3,"three")]
+
+    -- Check if we know the english words for 1 and 4
+    Map.member nums 1
+    > True
+
+    Map.member nums 4
+    > False
+
+
+    -- Get the english word for the number 3 and 4.
+    Map.lookup 3 nums
+    > Just "three"
+
+    Map.lookup 4 nums
+    > Nothing
+
+
+    -- Add (4, "four") to our original map.
+    let moreNums = Map.insert 4 "four" nums
+
+    Map.member moreNums 4
+    > True
+
+
+    -- Remove the entry for 1 from our original map.
+    let fewerNums = Map.delete 1 nums
+
+    Map.toAscList fewerNums
+    > [(2,"two"),(3,"three")]
+
+
+    -- Create a new map and combine it with our original map.
+    -- Note that these operations are left-biased.
+    let newNums = Map.fromList [(3,"new three"), (4,"new four")]
```

Comment:
Specify that `union` (not "these operations") is left-biased. The other operation here is `fromList`, which is basically *right-biased*. By the way, no, that biasing thing is *not my fault*. And it would probably be good to explain what the "left-biased" thing means.

Reply:
Done.